ui/main.py

```diff
@@ -65,6 +65,11 @@ def run(driver, config):
     store.dispatch(actions.trigger())
     button_loop(driver)
 
+    state    = store.get_state()
+    if state['update_ui'] == 'in progress':
+        store.dispatch(actions.update_ui('done')) # set as done, if it fails, need to be able to try again
```

Comment:
How does that stop us from calling it 'failed' and possibly making use of that info later?